powerauth-java-cmd/src/main/java/io/getlime/security/powerauth/app/cmd/Application.java

```diff
@@ -960,4 +977,40 @@ public void checkServerTrusted(java.security.cert.X509Certificate[] certs, Strin
 
     }
 
+    /**
+     * Read the contents of a provided data file.
+     * @param fileName File name.
+     * @param stepLogger Logger for error handling.
+     * @return Bytes with the contents of the file.
+     * @throws IOException In case reading the file failed.
+     * @throws ExecutionException In case the filename is null or a file does not exist.
+     */
+    private static byte[] readDataFile(String fileName, JsonStepLogger stepLogger) throws IOException, ExecutionException {
+        // check if the file was provided
+        if (fileName == null) { // filename was not provided, we are assuming empty data and log a warning
+            stepLogger.writeItem(
+                    "generic-warning-request-data-empty",
+                    "Empty request data file",
+                    "Request data file not provided, assuming empty data.",
+                    "WARNING",
+                    fileName
+            );
+            return new byte[0];
+        }
+
+        // Read data input file
+        if (Files.exists(Paths.get(fileName))) {
```

Comment:
We check that the file exists but do not check whether the file is readable, is that ok?

Reply:
This is a good catch - we had this issue everywhere. I replaced `Files.exists` with `Files.isReadable` which also checks if the file exists along the way...

Reply:
Ok, cool.